### PR TITLE
Fix failing RSpec tests

### PR DESCRIPTION
**Pull request by slorek**
## Description
This fixes two intermittently failing RSpec tests.

They previously failed intermittently because the database was not clean and sometimes the wrong object was unwittingly being referenced due to data conflicts.

In fact, neither of these tests should have passed if run properly in isolation, so previous passes were false positives.

I've added the [Database Cleaner gem](https://github.com/DatabaseCleaner/database_cleaner) to ensure RSpec tests are cleaned up properly after each test. Both tests failed reliably after adding this.

I also changed the CI environment setup so the unrelated test data seeds are no longer loaded; this data is unrelated to the automated tests and is intended for manual testing/demo purposes.

Because this was causing failures, I also refactored the `User#display_name` method to better reflect current business logic and to remove the assumption that `User.current` is set. This is not true in the RSpec suite. In the `test` suite they have [manually set this](https://github.com/UKGovernmentBEIS/beis-opss-psd/blob/master/psd-web/test/test_helper.rb#L78) in every test, but I don't think this is a good solution and exposes poor design. Instead, the 'current' user can be injected into the method as an argument. I also added instance variable caching to this method to hopefully improve performance; it is responsible for a lot of database queries currently according to Scout.

## Checklist:

- [x] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: psd-web/spec/rails_helper.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@
   # If you're not using ActiveRecord, or you'd prefer not to run each of your
   # examples within a transaction, remove the following line or assign false
   # instead of true.
-  config.use_transactional_fixtures = true
+  # config.use_transactional_fixtures = true
```
Comment (StupidCodeFactory) [inline psd-web/spec/rails_helper.rb]: we do want to user transactional fixtures

File: psd-web/spec/support/database_cleaner.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+RSpec.configure do |config|
+  config.before(:suite) do
```
Comment (StupidCodeFactory) [inline psd-web/spec/support/database_cleaner.rb]: this shouldn't be needed in rails 5.2

I'm wondering whether the problem doesn't come from the minitests fixtures

`https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/FixtureSet.html`
```ruby
class FooTest < ActiveSupport::TestCase
  self.use_transactional_fixtures = false
end
```

**Comment by StupidCodeFactory** [comment]
This is a quick fix and I think we should fix those properly- rails' team made a lot of effort to make the use of transaction work accross the board whether in system test or simple 
unit tests

File: psd-web/spec/support/database_cleaner.rb (hunk 2 above)
Comment (slorek) [reply psd-web/spec/support/database_cleaner.rb]: It's definitely coming from the RSpec suite. I didn't run the minitest suite between runs when testing the problem.

File: psd-web/spec/rails_helper.rb (hunk 1 above)
Comment (slorek) [reply psd-web/spec/rails_helper.rb]: It conflicts with database_cleaner - we get 19 failures with it set to `true`.

**Comment by slorek** [comment]
I did a bit of reading into the RSpec implementation and there are some limitations in the way it cleans up data using transactional fixtures only. I'm not sure it's feasible but I'd be happy if we can make it work later.

**Pull request merged**